orangecontrib/spectroscopy/widgets/owfft.py

```diff
@@ -532,35 +559,89 @@ def determine_sweeps(self):
 
     def check_metadata(self):
         """ Look for laser wavenumber and sampling interval metadata """
+       
+        # if self.data.attributes is not None:
         try:
-            lwn, _ = self.data.get_column_view("Effective Laser Wavenumber")
-        except ValueError:
-            if not self.dx_HeNe_cb.isEnabled():
-                # Only reset if disabled by this code, otherwise leave alone
-                self.dx_HeNe_cb.setDisabled(False)
-                self.infoc.setText("")
-                self.dx_HeNe = True
-                self.dx_edit.setDisabled(self.dx_HeNe)
-                self.dx = 1.0 / self.laser_wavenumber / 2.0
-            return
-        else:
-            lwn = lwn[0] if (lwn == lwn[0]).all() else ValueError()
+            self.reader = self.data.attributes['Reader']
+        except KeyError:
+            pass
+        if self.reader == 'NeaReaderGSF':
             self.dx_HeNe = False
-            self.dx_HeNe_cb.setDisabled(True)
-            self.dx_edit.setDisabled(True)
-        try:
-            udr, _ = self.data.get_column_view("Under Sampling Ratio")
-        except ValueError:
-            udr = 1
-        else:
-            udr = udr[0] if (udr == udr[0]).all() else ValueError()
+            self.dx_HeNe_cb.setDisabled(True) #??? Is possible change the label?
+            self.controls.auto_sweeps.setDisabled(True)
+            self.controls.sweeps = False
+            self.controls.peak_search.setEnabled(True)
+            self.controls.peak_search_enable.setDisabled(True)
+            self.controls.zpd1.setDisabled(True)
+            self.controls.zpd2.setDisabled(True)
+            self.controls.phase_corr.setDisabled(True)
+            self.controls.phase_res_limit.setDisabled(True)
+            self.controls.phase_resolution.setDisabled(True)         
+
+            info = self.data.attributes
+            number_of_points = int(info['Pixel Area (X, Y, Z)'][3])
+            scan_size = float(info['Interferometer Center/Distance'][2].replace(',', '')) #Microns
+            scan_size = scan_size*1e-4 #Convert to cm
+            OPD = scan_size * 2
+            step_size = OPD / (number_of_points - 1)
+            OPD = scan_size * 2
+
+            self.dx = step_size
+            self.sweeps = 0
+            self.phase_corr = 0
+            self.limit_output = False
+            self.apod_func = 0
+            self.zff = 2 #Because is power of 2
 
-        self.dx = (1 / lwn / 2 ) * udr
-        self.infoc.setText("{0} cm<sup>-1</sup> laser, {1} sampling interval".format(lwn, udr))
+        else:
```

Comment:
This `else` should be used to enable any disabled controls that aren't already restored if necessary. Test by connecting the Neagsf file, and then connecting `IFG_single`. The gui should be restored to full operation.

Since you will return after the if block above, the original code below doesn't need to be indented.

Reply:
I can `.setDisable (True)` some controls, so the code below needs `.setDisable (True)` the controls it needs.
I think you need to do this, because I don't understand all the processing of your data.